docs/source/guides/speed.rst

```diff
@@ -0,0 +1,301 @@
+
+
+Stop training when metric is degrading
+======================================
+The
+:class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping`
+callback can be used to monitor a validation metric and stop the training when no improvement is observed.
+
+.. testcode::
+
+    # 1. Import EarlyStopping callback
+    from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+
+    # 2. Log the metric you want to monitor using `self.log` method.
+    def validation_step(self, batch, batch_idx):
+        x, y = batch
+        z = self.encoder(x)
+        x_hat = self.decoder(z)
+        loss = F.mse_loss(x_hat, x)
+
+        self.log('val_loss', loss)
+
+    # 3. Init the callback, and set `monitor` to the logged metric of your choice.
+    early_stop_callback = EarlyStopping(monitor='val_loss');
+
+    # 4. Pass the callback to the Trainer `callbacks` flag.
+    trainer = Trainer(callbacks=[early_stop_callback])
+
+
+Stop training based on metric value
+===================================
+You can set a `stopping_threshold` to stop training immediately once the monitored quantity reaches this threshold. It is useful when we know that going beyond a certain optimal value does not further benefit us.
+
+.. testcode::
+
+    early_stop_callback = EarlyStopping(
+       monitor='val_accuracy',
+       stopping_threshold=0.98
+
+    )
+    trainer = Trainer(callbacks=[early_stop_callback])
+
+You can set a `divergence_threshold` to stop training as soon as the monitored quantity is lower than this threshold. When reaching a value this bad, we believe the model cannot recover anymore and it is better to stop early and run with different initial conditions.
+
+You can also set `check_finite` to stop training when the monitored metric becomes NaN or infinite.
+
+Learn more in the :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping` doc.
+
+
+Stopping an epoch early
+=======================
+
+You can stop an epoch early by overriding :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_start` to return ``-1`` when some condition is met.
+
+If you do this repeatedly, for every epoch you had originally requested, then this will stop your entire run.
+
+----------
+
+****************
+GPU/TPU training
+****************
+
+**Use when:** Running large datasets or want to speed up your training.
+
+With Lightning, running on GPUs, TPUs or multiple node is a simple switch of a flag.
+
+GPU training
+============
+
+Lightning supports a variety of plugins to further speed up distributed GPU training. Most notably:
+
+* :class:`~pytorch_lightning.plugins.training_type.DDPPlugin`
+* :class:`~pytorch_lightning.plugins.training_type.DDPShardedPlugin`
+* :class:`~pytorch_lightning.plugins.training_type.DeepSpeedPlugin`
+
+.. code-block:: python
+
+    # run on 1 gpu
+    trainer = Trainer(gpus=1)
```

Comment:
`accelerator="ddp"` is not mentioned here, but considering this document is about speed then ddp is definitely the preferred default plugin and I think should be mentioned.

Reply:
is it not the default?